pkg/reconciler/apiresource/controller.go

```diff
@@ -0,0 +1,373 @@
+		},
+	})
+	c.negotiatedApiResourceLister = apiresourceSif.Apiresource().V1alpha1().NegotiatedAPIResources().Lister()
+	apiresourceSif.Apiresource().V1alpha1().APIResourceImports().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.enqueue(Add, nil, obj) },
+		UpdateFunc: func(oldObj, obj interface{}) { c.enqueue(Update, oldObj, obj) },
+		DeleteFunc: func(obj interface{}) { c.enqueue(Delete, nil, obj) },
+	})
+	c.apiResourceImportIndexer = apiresourceSif.Apiresource().V1alpha1().APIResourceImports().Informer().GetIndexer()
+	c.apiResourceImportIndexer.AddIndexers(map[string]cache.IndexFunc{
+		ClusterNameAndGVRIndexName: func(obj interface{}) ([]string, error) {
+			if apiResourceImport, ok := obj.(*apiresourcev1alpha1.APIResourceImport); ok {
+				return []string{GetClusterNameAndGVRIndexKey(apiResourceImport.ClusterName, apiResourceImport.GVR())}, nil
+			}
+			return []string{}, nil
+		},
+	})
+	c.apiResourceImportLister = apiresourceSif.Apiresource().V1alpha1().APIResourceImports().Lister()
+
+	apiresourceSif.WaitForCacheSync(stopCh)
+	apiresourceSif.Start(stopCh)
+
+	crdSif := crdexternalversions.NewSharedInformerFactoryWithOptions(apiextensionsclient.NewForConfigOrDie(cfg), resyncPeriod)
+	crdSif.Apiextensions().V1().CustomResourceDefinitions().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.enqueue(Add, nil, obj) },
+		UpdateFunc: func(oldObj, obj interface{}) { c.enqueue(Update, oldObj, obj) },
+		DeleteFunc: func(obj interface{}) { c.enqueue(Delete, nil, obj) },
+	})
+	c.crdIndexer = crdSif.Apiextensions().V1().CustomResourceDefinitions().Informer().GetIndexer()
+	c.crdIndexer.AddIndexers(map[string]cache.IndexFunc{
+		ClusterNameAndGVRIndexName: func(obj interface{}) ([]string, error) {
+			if crd, ok := obj.(*apiextensionsv1.CustomResourceDefinition); ok {
+				return []string{GetClusterNameAndGVRIndexKey(crd.ClusterName, metav1.GroupVersionResource{
+					Group:    crd.Spec.Group,
+					Resource: crd.Spec.Names.Plural,
+				})}, nil
+			}
+			return []string{}, nil
+		},
+	})
+	c.crdLister = crdSif.Apiextensions().V1().CustomResourceDefinitions().Lister()
+	crdSif.WaitForCacheSync(stopCh)
+	crdSif.Start(stopCh)
+
+	return c
+}
+
+type Controller struct {
+	queue workqueue.RateLimitingInterface
+
+	apiresourceClient            typedapiresource.ApiresourceV1alpha1Interface
+	negotiatedApiResourceIndexer cache.Indexer
+	negotiatedApiResourceLister  apiresourcelister.NegotiatedAPIResourceLister
+	apiResourceImportIndexer     cache.Indexer
+	apiResourceImportLister      apiresourcelister.APIResourceImportLister
+
+	crdClient  typedapiextensions.ApiextensionsV1Interface
+	crdIndexer cache.Indexer
+	crdLister  crdlister.CustomResourceDefinitionLister
+
+	kubeconfig                       clientcmdapi.Config
+	stopCh                           chan struct{}
+	AutoPublishNegotiatedAPIResource bool
+}
+
+type QueueElementType string
```

Comment:
I think most of these types and enum values don't need to be exported.

Reply:
Fixed in commit 6a4dcd07a4a268ef7dd0158950e2a1c8452b9eb3